File: Python/PythonKatas/FizzBuzz/fizzbuzz.py
def FizzBuzz(value):
    result = ""
    if value <= 0:
        return result

    if value % 3 == 0:
        result += "fizz"
    if value % 5 == 0:
        result += "buzz"
    if not result:
        result += str(value)

    return result

def FizzBuzzExtended(value):
    result = ""
    if value <= 0:
        return result

    if value % 3 == 0 or "3" in str(value):
        result += "fizz"
    if value % 5 == 0 or "5" in str(value):
        result += "buzz"
    if not result:
        result += str(value)

    return result

File: TypeScript/TypeScriptKatas/FizzBuzz/fizzbuzz.ts
class FizzBuzz {

    doFizzBuzz(value: number): string {

        var result = "";
        if (value <= 0) return result;

        if (value % 3 === 0) result = "fizz";
        if (value % 5 === 0) result += "buzz";

        return !result ? value.toString() : result;
    }

    doFizzBuzzExtended(value: number): string {

        var result = "";
        if (value <= 0) return result;
        var text = value.toString();

        if (value % 3 === 0 || text.search("3") !== -1) result = "fizz";
        if (value % 5 === 0 || text.search("5") !== -1) result += "buzz";

        return !result ? text : result;
    }

}